plugins/http_plugin/http_plugin.cpp

```diff
@@ -271,6 +271,7 @@ class http_plugin_impl : public std::enable_shared_from_this<http_plugin_impl> {
 
    void http_plugin::plugin_initialize(const variables_map& options) {
       try {
+         handle_sighup(); // setup logging
```

Comment:
Not related to your change, so maybe ignore, but:

- no log message "Initializing HTTP plugin" here (same for other plugins)?
- why are these functions named `handle_sigup`, when they initialize the logger?
- inside `handle_sigup` we copy and note that a copy is needed while `name` is passed as a const reference to logger::update()

Reply:
1. One could argue it is not needed in net_plugin either since no one ever noticed it was not there. Likely it is noise unless you are trying to debug something. I left it in because "why not"?
2. They are named `handle_sigup` because they override the `handle_sigup` of virtual `appbase::plugin::handle_sighup` which is called anytime a `SIGHUP` is caught. Specifically in Leap we wish to update the logging configuration on `SIGHUP`. We have talked about the possibility of updating other configuration params on `SIGHUP` as well. 
We could create a `update_logger` method for all the plugins and have the override `handle_sigup` call that and also call it from `plugin_initialize`. The lazy way was chosen and just call `handle_sighup` because that is what is needed on startup.
3. Yes, the comment is correct. If you instead did `fc::logger::update(logger().name(), logger())` the `const&` bound to the internals of `logger()` can be destroyed by the `update` of the referenced `logger()`. `name` can then be used after `logger()` is updated causing access to delete memory.

Reply:
Thanks for the answers Kevin. 
regarding point 3, I think a  better (and safer) alternative would be to have `logger().name()` return a value and not a reference to a string whose lifetime is not clearly known

Reply:
I agree. I'll go ahead and fix that.